routing/payment_session.go

```diff
@@ -298,6 +298,15 @@ func (p *paymentSession) RequestRoute(maxAmt, feeLimit lnwire.MilliSatoshi,
 			// route, try it for half the amount.
 			maxAmt /= 2
 
+			// If this shard ends up being too large, then we'll
+			// clamp it down to the max shard size if that's set.
+			//
+			// TODO(roasbeef): extend the integrated routing tets w/ info
+			maxShardActive := p.payment.MaxShardAmt != nil
+			if maxShardActive && *p.payment.MaxShardAmt > maxAmt {
```

Comment:
Def, excellent observation!

Reply:
I think we actually only need to check it at the top of `RequestRoute` as well, since we'd clamp before entering the main loop, which ensures we won't ever try to send a payment above the shard size.